### PR TITLE
Maintenance/simplify env

### PR DESCRIPTION
**Pull request by millii19**
simplified env variables, nginx conf files and removed client-authentication

### CODE REVIEW

**Review by absurd-turtle — APPROVED** [review]
LGTM!
I had a few comments but nothing too important.
Good job stripping out unnecessary complexity and the auth stuff :)

**Review by absurd-turtle — CHANGES_REQUESTED** [review]
My local build doesn't seem to run correctly.
I get `An error occurred during a connection to localhost. PR_END_OF_FILE_ERROR` when I try to reach `https://localhost/`.
Is there something I missed?

**Comment by absurd-turtle** [comment]
> My local build doesn't seem to run correctly. I get `An error occurred during a connection to localhost. PR_END_OF_FILE_ERROR` when I try to reach `https://localhost/`. Is there something I missed?

Connecting with http at `http://localhost/app/` works as intended though.

File: api/src/caching/memcached.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import Keyv from "keyv"
 import KeyvMemcache from "@keyv/memcache";
 import { KeyvAdapter } from "@apollo/utils.keyvadapter"
+import { MEMCACHED_HOST } from "../helpers/settings.js";
```
[the remark this answers is not in the document]
Comment (millii19) [reply api/src/caching/memcached.ts:4]: This is one of the limitations of typescript+ES modules, the imports need to be .js

File: api/src/helpers/settings.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export const ARANGO_HOST = process.env.ARANGODB_HOST ?? "arangodb"
+export const ARANGO_PORT = process.env.ARANGODB_PORT ?? 8529
+export const ARANGO_DB = process.env.ARANGODB_DATABASE ?? "OOT"
+export const ARANGO_USER = process.env.ARANGODB_USER
+export const getArangoPassword = () => process.env.ARANGODB_PASSWORD
+
+export const SPOTIFY_CLIENT_ID = process.env.SPOTIFY_CLIENT_ID
+export const SPOTIFY_CLIENT_SECRET = process.env.SPOTIFY_CLIENT_SECRET
+export const SPOTIFY_SCOPE = process.env.SPOTIFY_SCOPE
```
[the remark this answers is not in the document]
Comment (millii19) [reply api/src/helpers/settings.ts:15]: I'm working on an improvement for the environment settings, there I will add the default.

File: api/src/helpers/settings.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import dotenv from 'dotenv'
+dotenv.config()
+
+export const API_PORT = Number.parseInt(process.env.APOLLO_PORT ?? "3003")
```
[the remark this answers is not in the document]
Comment (millii19) [reply api/src/helpers/settings.ts:4]: Same, there will be nice error handling in the improvement

**Comment by millii19** [comment]
both http://localhost and http://localhost/app work for me, when I run the docker compose locally

**Review by absurd-turtle — APPROVED** [review]

**Pull request merged**